### ISSUE TITLE
New Rule: Prefer private class fields/methods - forbid starting with undescore

### ISSUE THREAD

**Issue by jimmywarting** [eslint/eslint#15417]
### Rule details

Prefer private class fields/methods

### Related ECMAScript feature

Private class fields

### What type of rule is this?

Enforces a formatting/stylistic preference

### Example code

should fail:

```js
class Message {
  constructor(message) {
    this._message = message // Warns about setting a soft private property
    this._scream() // Warns about calling a soft private method
  }

  _scream() { // Warns about not using private `#` instead
    alert(this._message) // Warns about accessing a soft private property
  }
}
```

Should be okey:
```js
class Message {
  constructor(message) {
    this.#message = message
    this.#scream()
  }

  #scream() {
    alert(this.#message)
  }
}
```

Or if it's typescript then fail on this:
```ts
class A {
  private message = ''
  private scream() { }
}
```

### Why should this rule be in the core instead of a plugin?

using `_` is usually an indication of a private function or a private field. Using `_` isn't truly private. and can have consequence if someone mess around with it and can break at any point.

this rule should ideally only be used when the targeted version supports private method/fields

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by nzakas** [comment]
Thanks for the suggestion. We aren’t adding any more rules that just enforce naming conventions. You can probably use no-restricted-syntax to flag these instances or else you can create a custom rule that does exactly what you want.
